plugins/core/src/main/kotlin/de/fayard/refreshVersions/core/internal/MavenDependencyVersionsFetcherGCS.kt

```diff
@@ -0,0 +1,71 @@
+package de.fayard.refreshVersions.core.internal
+
+import com.google.auth.oauth2.ServiceAccountCredentials
+import com.google.cloud.storage.Bucket
+import com.google.cloud.storage.StorageException
+import com.google.cloud.storage.StorageOptions
+import de.fayard.refreshVersions.core.ModuleId
+import org.gradle.api.GradleException
+import org.gradle.caching.BuildCacheException
+import java.io.FileInputStream
+import java.io.FileNotFoundException
+import java.io.IOException
+
+internal class MavenDependencyVersionsFetcherGCS(
+    moduleId: ModuleId,
+    repoUrl: String
+) : MavenDependencyVersionsFetcher(
+    moduleId = moduleId,
+    repoUrl = repoUrl
+) {
+    override suspend fun getMetadata(): String {
+        return getBlob("${moduleId.group!!.replace('.', '/')}/${moduleId.name}/maven-metadata.xml")
+    }
+
+    private val bucketName = repoUrl.substringAfter("gcs://").substringBefore("/")
+    private val repoPath = repoUrl.substringAfter("gcs://").substringAfter("/")
+
+    private val bucket: Bucket
+
+    init {
+        try {
+
+            val storage = StorageOptions.newBuilder()
+                .setCredentials(ServiceAccountCredentials.fromStream(FileInputStream(CREDENTIALS_PATH)))
+                .build()
+                .service
+
+            bucket = storage.get(bucketName) ?: throw BuildCacheException("$bucketName is unavailable")
+
+        } catch (e: FileNotFoundException) {
+            throw GradleException("Unable to load credentials from $CREDENTIALS_PATH.", e)
+        } catch (e: IOException) {
+            throw GradleException("Unable to access Google Cloud Storage bucket '$bucketName'.", e)
+        } catch (e: StorageException) {
+            throw GradleException("Unable to access Google Cloud Storage bucket '$bucketName'.", e)
+        }
+    }
+
+    @OptIn(ExperimentalStdlibApi::class)
+    fun getBlob(path: String): String {
+        val fullPath = listOfNotNull(repoPath, path).joinToString("/", "", "")
```

Comment:
The last two parameters are not needed because they are already the default values.
If they had to be defined, it'd be way better to use named arguments instead of throwing same typed arguments in there without any indication about what is what.

Reply:
Also, there's no need to use `listOfNotNull` because the arguments are not nullable… and using `joinToString` is overkill and obscures the code as there's only two `String`s to join, which means a string template would work as well, even a little more efficiently actually.